cpp/include/cudf/io/detail/parquet.hpp

```diff
@@ -211,5 +212,14 @@ class writer {
     std::vector<std::unique_ptr<std::vector<uint8_t>>> const& metadata_list);
 };
 
+/**
+ * @brief Reads metadata of parquet dataset.
+ *
+ * @param sources Dataset sources to read from
+ *
+ * @return parquet_metadata with parquet schema, number of rows, number of row groups and key-value
+ * metadata.
+ */
+parquet_metadata read_parquet_metadata(std::vector<std::unique_ptr<datasource>>&& sources);
```

Comment:
Do we want to make `read_parquet_metadata` uncallable with l-values (as it is now)? We don't really have to move from the sources, IMO const& is fine here.

Reply:
Replaced with const-ref.